include/range/v3/at.hpp

```diff
@@ -30,7 +31,20 @@ namespace ranges
         {
             /// \return `begin(rng)[n]`
             template<typename Rng,
-                CONCEPT_REQUIRES_(RandomAccessRange<Rng>())>
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  SizedRange<Rng>())>
+            RANGES_CXX14_CONSTEXPR
+            auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const ->
+                decltype(begin(rng)[n])
+            {
+                RANGES_EXPECT(static_cast<std::size_t>(n) < ranges::size(rng));
+                return begin(rng)[n];
+            }
+
+            /// \return `begin(rng)[n]`
+            template<typename Rng,
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  not SizedRange<Rng>())>
```

Comment:
@ericniebler I've fixed it.

@CaseyCarter I get an error if I remove the cast:

```shell
 error: comparison of integers of different signs: 'range_difference_type_t<int (&)[4]>'
      (aka 'long') and 'decltype(fn::impl_(r, 42))' (aka 'unsigned long')
```